packages/scripts/src/admin-proposals/simulateVoteV2.ts

```diff
@@ -0,0 +1,246 @@
+// Description:
+// - Simulate voting affirmatively on any pending Admin Proposals
+
+import { advanceBlockAndSetTime, computeVoteHashAncillary, getRandomSignedInt, getWeb3ByChainId } from "@uma/common";
+import { GovernorV2Ethers, VotingTokenEthers, VotingV2Ethers } from "@uma/contracts-node";
+import { BigNumberish, BytesLike } from "ethers";
+// import { REQUIRED_SIGNER_ADDRESSES, SECONDS_PER_DAY, YES_VOTE } from "../utils/constants";
+import { getContractInstance } from "../utils/contracts";
+
+const SECONDS_PER_DAY = 86400;
+const YES_VOTE = "1";
+const REQUIRED_SIGNER_ADDRESSES = {
+  deployer: "0x2bAaA41d155ad8a4126184950B31F50A1513cE25",
+  foundation: "0x7a3A1c2De64f20EB5e916F40D11B01C441b2A8Dc",
+  account_with_uma: "0xcb287f69707d84cbd56ab2e7a4f32390fa98120b",
+};
+
+// Run:
+// - Check out README.md in this folder for setup instructions and simulating votes between the Propose and Verify
+//   steps.
+// - This script should be run after any Admin proposal UMIP script against a local Mainnet fork. It allows the tester
+//   to simulate what would happen if the proposal were to pass and to verify that contract state changes as expected.
+// - Vote Simulate: NODE_URL_1=http://localhost:9545 node ./packages/scripts/src/admin-proposals/simulateVoteV2.js --network mainnet-fork
+
+const hre = require("hardhat");
+const { toWei } = hre.web3.utils;
+require("dotenv").config();
+const assert = require("assert").strict;
+
+async function simulateVoteV2() {
+  const foundationSigner = await hre.ethers.getSigner(REQUIRED_SIGNER_ADDRESSES["foundation"]);
+
+  const governorV2Address = process.env["GOVERNOR_V2_ADDRESS"];
+  const votingV2Address = process.env["VOTING_V2_ADDRESS"];
+
+  const governorV2 = await getContractInstance<GovernorV2Ethers>("GovernorV2", governorV2Address);
+  const votingV2 = await getContractInstance<VotingV2Ethers>("VotingV2", votingV2Address);
+  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
+
+  const web3 = getWeb3ByChainId(1);
+  const accounts = await web3.eth.getAccounts();
```

Comment:
this is a little bit janky. we are using _both_ web3 and ethers in this script. it would be better to just use one or the other, ideally. I think you might need to move to just using web3, unfortunately, seeing some methods, like `advanceBlockAndSetTime` are only configured to work with web3. else, you'll need to make this work with ethers as well.

Reply:
[Fixed](https://github.com/UMAprotocol/protocol/pull/4190/commits/7fd6601f32da154b095fe49af95329d1ed943b56)